metals/src/main/scala/scala/meta/internal/implementation/SuperMethodProvider.scala

```diff
@@ -1,295 +1,27 @@
-        value
-    }
-
-  }
-
-  private def calculateClassSuperHierarchy(
-      classSymbolInformation: SymbolInformation,
-      findSymbol: String => Option[SymbolInformation]
-  ): List[ClassHierarchyItem] = {
-    getSuperClasses(
-      classSymbolInformation,
-      findSymbol,
-      m.Set[String](),
-      Map()
-    )
-  }
-
-  private def findSuperForMethodOrFieldChecked(
-      msi: SymbolInformation,
-      methodInfo: MethodSignature,
-      documentWithPath: TextDocumentWithPath,
-      cache: LensGoSuperCache,
-      findSymbol: String => Option[SymbolInformation]
-  ): Option[String] = {
-    val classSymbolInformationOption =
-      findClassInfo(
-        msi.symbol,
-        msi.symbol.owner,
-        documentWithPath.textDocument,
-        findSymbol
-      )
-    val methodName = msi.displayName
-    val result = for {
-      classSymbolInformation <- classSymbolInformationOption.toIterable
-      bottomClassSig =
-        classSymbolInformation.signature
-          .asInstanceOf[ClassSignature]
-      ClassHierarchyItem(superClass, asSeenFrom) <-
-        calculateClassSuperHierarchyWithCache(
-          classSymbolInformation,
-          cache,
-          findSymbol
-        )
-      classSig <- List(superClass.signature).collect {
-        case cs: ClassSignature => cs
-      }
-      methodSymbolLink <- classSig.getDeclarations.symlinks
-      if msi.isLocal || methodSymbolLink.contains(methodName)
-      mSymbolInformation <- findSymbol(methodSymbolLink).toIterable
-      methodSignature <- Option(mSymbolInformation.signature).collect {
-        case m: MethodSignature => m
-      }
-      translatedASF = AsSeenFrom.toRealNames(
-        classSig,
-        bottomClassSig,
-        Some(asSeenFrom)
-      )
-      if checkSignaturesEqual(
-        mSymbolInformation,
-        methodSignature,
-        msi,
-        methodInfo,
-        translatedASF,
-        findSymbol
-      )
-    } yield mSymbolInformation.symbol
-    result.headOption
+      methodSymbolInformation: SymbolInformation
+  ): List[String] = {
+    methodSymbolInformation.overriddenSymbols.filter { sym =>
+      !stopSymbols.exists(stop => sym.startsWith(stop))
+    }.toList
   }
 
-  private def isDefinitionOfMethodField(
-      symbolRole: SymbolOccurrence.Role,
-      symbolInformation: SymbolInformation
-  ): Option[MethodSignature] =
-    symbolInformation.signature match {
-      case methodSignature: MethodSignature
-          if symbolRole.isDefinition && (symbolInformation.isMethod || symbolInformation.isField) =>
-        Some(methodSignature)
-      case _ =>
-        None
-    }
-
-  final val stopSymbols: Set[String] = Set(
-    "scala/AnyRef#",
-    "scala/Serializable#",
-    "java/io/Serializable#",
-    "scala/AnyVal#",
-    "scala/Any#"
+  private final val stopSymbols: Set[String] = Set(
+    "scala/AnyRef#", "scala/Serializable#", "java/io/Serializable#",
+    "java/lang/Object#", "scala/AnyVal#", "scala/Any#"
```

Comment:
Added `"java/lang/Object#"` because `toString` method of case classes overrides `java/lang/Object#toString()`